workspace/__ardulib__/Graphics/XGraphics.h

```diff
@@ -0,0 +1,458 @@
+
+#ifndef X_GRAPHICS_H
+#define X_GRAPHICS_H
+
+#include "XRender.h"
+#include "XVector2.h"
+
+using xod::XColor;
+
+class XFill {
+private:
+public:
+    XFill() {}
+    virtual XColor colorAtPoint(XVector2<int16_t> point) {}
+    virtual void setSolidColor(XColor _color) {}
+};
+
+class XSolidFill : public XFill {
+private:
+    XColor _color;
+public:
+    XSolidFill(XColor color) {
```

Comment:
This file is way too big. Would you split it into h+cpp pair per concept? I see the following:

- Fill
- Stroke (for now, I’m pretty sure this is an extraneous concept)
- Bitmap
- Canvas
- Point
- Line
- Rectangle

Reply:
But I'm not sure about Stroke, because it's pretty much simpler to set stroke instead of creating another one figure (especially, for curves or even for diagonal lines). :-P